packages/blocks/src/_common/theme/theme-observer.ts

```diff
@@ -35,6 +42,48 @@ export class ThemeObserver extends Slot<CssVariablesMap> {
     this._observer?.disconnect();
   }
 
+  // Generates a CSS's color property.
+  generateColorProperty(color: string | CustomColor, fallback: string) {
+    color = this.getColor(color, fallback);
+    return color.startsWith('--') ? `var(${color})` : color;
+  }
+
+  // Generates a CSS's color property with `light-dark` function.
+  // Usually used to set the frame/note background.
+  generateColorPropertyWithLightDark(
+    color: string | CustomColor,
+    fallback: string
+  ) {
+    fallback = fallback.startsWith('--') ? `var(${fallback})` : fallback;
+
+    if (typeof color === 'string') {
+      return color.startsWith('--')
+        ? `var(${color}, ${fallback})`
+        : color ?? fallback;
+    }
+
+    if (color.light && color.dark) {
+      return `light-dark(${color.light}, ${color.dark})`;
+    }
+
+    return color.normal ?? fallback;
+  }
+
+  // Gets a color without going through computed style.
+  // e.g. `rgba(255,0,0)` `#fff` `--affine-palette-shape-blue`
+  getColor(color: string | CustomColor, fallback: string) {
```

Comment:
These new methods are somewhat redundant in that they provide very close functionality. They are fully controllable through optional parameters.